interfaces/builtin/snap_interfaces_requests_control.go

```diff
@@ -35,13 +41,46 @@ const snapPromptingControlBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
+type requestsControlInterface struct {
+	commonInterface
+}
+
+func (iface *requestsControlInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	// snaps can declare "handler-service" attribute on the plug, which
+	// indicates which user service will be handling prompt requests
+
+	handlerServiceAttr, isSet := plug.Attrs["handler-service"]
+	handlerService, ok := handlerServiceAttr.(string)
+	if isSet && !ok {
+		return fmt.Errorf(`snap-interfaces-requests-control "handler-service" attribute must be a string, not %v`,
+			handlerServiceAttr)
+	}
+
+	if handlerService == "" {
+		return nil
+	}
+
+	svc := plug.Snap.Apps[handlerService]
+	if svc == nil {
+		return fmt.Errorf("declared handler service %q not found in snap", handlerService)
+	}
+
+	if !svc.IsService() || svc.DaemonScope != snap.UserDaemon {
+		return fmt.Errorf("declared handler service %q is not a user daemon", handlerService)
```

Comment:
For consistency with the error above.

```suggestion
		return fmt.Errorf("declared handler service %q is not a user service", handlerService)
```